roachtest: add inconsistency test

tbg opened the pull request:
This commit adds a new roachtest 'inconsistency', which starts up a
three-node cluster, waits for replication, stops it, introduces an
inconsistency on the r1 replica on n1, restarts the cluster, and
asserts that n1 gets terminated by the consistency checker (it is
randomly either the leaseholder or not at this point).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42264)
<!-- Reviewable:end -->

bdarnell reviewed (approved):
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42264)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @irfansharif)



<!-- Sent from Reviewable.io -->

irfansharif reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42264#-:-LtAnyCl16TcP9SUVCI4:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/cmd/roachtest/inconsistency.go, line 112 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/42264#-LtAnpfg0cWqGdrcytyG:-LtAnpfg0cWqGdrcytyH:b-fz37vo) ([raw file](https://github.com/cockroachdb/cockroach/blob/33ca0a1af44c8ba3354b8d5acabfd8f4348aa4d6/pkg/cmd/roachtest/inconsistency.go#L112)):*
> ```Go
> 		t.Fatalf("expected one dead NodeID, got %v", ids)
> 	}
> 	c.Run(ctx, c.Node(1), "grep "+""+
> ```

What's the `""` for? And why all the "."?


<!-- Sent from Reviewable.io -->

tbg reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42264)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @irfansharif)

___
*[pkg/cmd/roachtest/inconsistency.go, line 112 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/42264#-LtAnpfg0cWqGdrcytyG:-LtPBqSk4JWhP3zKI3cN:b-n5g48h) ([raw file](https://github.com/cockroachdb/cockroach/blob/33ca0a1af44c8ba3354b8d5acabfd8f4348aa4d6/pkg/cmd/roachtest/inconsistency.go#L112)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

What's the `""` for? And why all the "."?
</blockquote></details>

The `+""+` is Goland having played a trick on me (fixed). The rest is because `c.Run` notoriously has trouble with whitespace and so I prefer to not hand it any.


<!-- Sent from Reviewable.io -->

irfansharif reviewed (approved):
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42264)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: add inconsistency test

This commit adds a new roachtest 'inconsistency', which starts up a
three-node cluster, waits for replication, stops it, introduces an
inconsistency on the r1 replica on n1, restarts the cluster, and
asserts that n1 gets terminated by the consistency checker (it is
randomly either the leaseholder or not at this point).

Release note: None

tbg commented:
I added an end-to-end-test for #40442 into this roachtest and it is now skipped for `<= 19.2.1` (the backports will be in 19.2.2).

TFTRs!

bors r=irfansharif,bdarnell
